server/repository/src/db_diesel/vvm_status/vvm_status_log_row.rs

```diff
@@ -62,28 +62,49 @@ impl<'a> VVMStatusLogRowRepository<'a> {
         &self,
         line_id: &str,
     ) -> Result<Vec<VVMStatusLogRow>, RepositoryError> {
-        let result = vvm_status_log::table
+        let result = vvm_status_log
             .filter(vvm_status_log::stock_line_id.eq(line_id))
             .order(vvm_status_log::created_datetime.desc())
             .load(self.connection.lock().connection())?;
         Ok(result)
     }
 
-    pub fn upsert_one(&self, row: &VVMStatusLogRow) -> Result<(), RepositoryError> {
-        diesel::insert_into(vvm_status_log::table)
+    pub fn _upsert_one(&self, row: &VVMStatusLogRow) -> Result<(), RepositoryError> {
+        diesel::insert_into(vvm_status_log)
             .values(row)
             .on_conflict(id)
             .do_update()
             .set(row)
             .execute(self.connection.lock().connection())?;
         Ok(())
     }
+
+    pub fn upsert_one(&self, row: &VVMStatusLogRow) -> Result<i64, RepositoryError> {
+        self._upsert_one(row)?;
+        self.insert_changelog(row, RowActionType::Upsert)
+    }
```

Comment:
Nit: Don't need to do this separately? 

```suggestion
    pub fn upsert_one(&self, row: &VVMStatusLogRow) -> Result<i64, RepositoryError> {
        diesel::insert_into(vvm_status_log)
            .values(row)
            .on_conflict(id)
            .do_update()
            .set(row)
            .execute(self.connection.lock().connection())?;
        self.insert_changelog(row, RowActionType::Upsert)
    }
```